### ISSUE TITLE
429 responses lack Access-Control-Allow-Origin header

### ISSUE THREAD

**Issue by zBNF** [CryptoBlades/cryptoblades#484]
When a user triggers a rate-limit, 429 responses lack Access-Control-Allow-Origin preventing the browser from returning the server response.

Blocks: #477 

---

Will make a PR to fix this shortly.

**Comment by seiyria** [comment]
At present, this was done but reverted as it caused some issues. @crypto-smoke ran into them earlier, I think it was blocking some server calls. See https://github.com/CryptoBlades/cryptoblades-api/commit/bda400f56cb1e3a2043223dea55197a9ac4cdc56 for the specific code involved.

If this were to be re-added, ideally it would be bypassable with the authentication middleware checking for the secret in the header.

**Comment by zBNF** [comment]
See https://github.com/CryptoBlades/cryptoblades-api/pull/6

**Comment by zBNF** [comment]
This is a separate issue from the one you're describing. This issue is about how requests that are blocked by the rate-limiter never get the `Access-Control-Allow-Origin: '*'` header set, thus preventing js from being able to read the response. I'm fairly sure this is a blocking issue to #477.

What you're referring to is setting `Access-Control-Allow-Origin: '*'` to a less permissive value. I think the issue you described above would be expected from that code as it will force the server to throw an exception when the request does not include an `Origin` header that matches `cryptoblades.io`. @crypto-smoke would have needed to add an Origin header with a value that matches the regex check for the request to succeed.

**Comment by zBNF** [comment]
I created #487 to take and submitted a PR to take a stab at tightening Access-Control-Allow-Origin.

**Comment by crypto-smoke** [comment]
Believe this should be closed due to merge of https://github.com/CryptoBlades/cryptoblades-api/pull/6

**Comment by seiyria** [comment]
@zBNF since the repository is different, you'll have to move this issue into bounty requested. I will add you to the github org so you can manage future bounties.
